modules/evm/src/lib.rs

```diff
@@ -214,30 +219,25 @@ decl_module! {
 		) -> DispatchResultWithPostInfo {
 			T::CallOrigin::ensure_address_origin(&source, origin)?;
 
-			match T::Runner::call(
-				source,
-				target,
-				input,
-				value,
-				gas_limit,
-			)? {
-				CallInfo {
-					exit_reason: ExitReason::Succeed(_),
-					..
-				} => {
-					Module::<T>::deposit_event(Event::<T>::Executed(target));
-				},
-				info => {
-					Module::<T>::deposit_event(Event::<T>::ExecutedFailed(target, info.exit_reason, info.output));
-				},
+			let info = T::Runner::call(source, target, input, value, gas_limit)?;
+
+			if info.exit_reason.is_succeed() {
+				Module::<T>::deposit_event(Event::<T>::Executed(target));
+			} else {
+				Module::<T>::deposit_event(Event::<T>::ExecutedFailed(target, info.exit_reason, info.output));
 			}
 
-			Ok(Pays::No.into())
+			let used_gas: u32 = info.used_gas.unique_saturated_into();
+
+			Ok(PostDispatchInfo {
+				actual_weight: Some(T::GasToWeight::convert(used_gas)),
+				pays_fee: Pays::No
```

Comment:
```suggestion
				pays_fee: Pays::Yes
```

We want to use our custom fee handling logic to handle the tx fee instead of evm gas fee